cli/command/disks/commit.go

```diff
@@ -0,0 +1,198 @@
+	comm "github.com/opencurve/curveadm/internal/common"
+	"github.com/opencurve/curveadm/internal/configure/disks"
+	"github.com/opencurve/curveadm/internal/configure/hosts"
+	"github.com/opencurve/curveadm/internal/errno"
+	"github.com/opencurve/curveadm/internal/storage"
+	tui "github.com/opencurve/curveadm/internal/tui/common"
+	"github.com/opencurve/curveadm/internal/utils"
+	"github.com/spf13/cobra"
+)
+
+const (
+	COMMIT_EXAMPLE = `Examples:
+  $ curveadm disks commit /path/to/disks.yaml  # Commit disks`
+)
+
+type commitOptions struct {
+	filename string
+	slient   bool
+}
+
+func NewCommitCommand(curveadm *cli.CurveAdm) *cobra.Command {
+	var options commitOptions
+
+	cmd := &cobra.Command{
+		Use:     "commit DISKS [OPTIONS]",
+		Short:   "Commit disks",
+		Args:    utils.ExactArgs(1),
+		Example: COMMIT_EXAMPLE,
+		RunE: func(cmd *cobra.Command, args []string) error {
+			options.filename = args[0]
+			return runCommit(curveadm, options)
+		},
+		DisableFlagsInUseLine: true,
+	}
+
+	flags := cmd.Flags()
+	flags.BoolVarP(&options.slient, "slient", "s", false, "Slient output for disks commit")
+
+	return cmd
+}
+
+func readAndCheckDisks(curveadm *cli.CurveAdm, options commitOptions) (string, error) {
+	// 1) read disks from file
+	if !utils.PathExist(options.filename) {
+		return "", errno.ERR_DISKS_FILE_NOT_FOUND.
+			F("%s: no such file", utils.AbsPath(options.filename))
+	}
+	data, err := utils.ReadFile(options.filename)
+	if err != nil {
+		return data, errno.ERR_READ_DISKS_FILE_FAILED.E(err)
+	}
+
+	// 2) display difference
+	oldData := curveadm.Disks()
+	if !options.slient {
+		diff := utils.Diff(oldData, data)
+		curveadm.WriteOutln(diff)
+	}
+
+	// 3) check disks data
+	_, err = disks.ParseDisks(data)
+	return data, err
+}
+
+func updateDisk(data string, curveadm *cli.CurveAdm) error {
+	var err error
+	hcs, err := hosts.ParseHosts(curveadm.Hosts())
+	if err != nil {
+		return err
+	}
+	dcs, err := disks.ParseDisks(data)
+	if err != nil {
+		return err
+	}
+
+	keySep := ":"
+	disksToSync := make(map[string]bool)
+
+	// write disk records
+	for _, hc := range hcs {
+		for _, dc := range dcs {
+			host := hc.GetHost()
+			device := dc.GetDevice()
+			diskOnlyHosts := utils.InterfaceSlice2Map(dc.GetHostsOnly())
+			diskExcludeHosts := utils.InterfaceSlice2Map(dc.GetHostsExclude())
+			if _, ok := diskOnlyHosts[host]; !ok && len(diskOnlyHosts) > 0 {
+				continue
+			}
+			if _, ok := diskExcludeHosts[host]; ok {
+				continue
+			}
+
+			disksToSync[strings.Join([]string{host, device}, keySep)] = true
+
+			if diskRecords, err := curveadm.Storage().GetDisk(
+				comm.DISK_DEVICE, host, device); err != nil {
+				return err
+			} else if len(diskRecords) > 0 {
+				continue
+			}
+
+			if err := curveadm.Storage().SetDisk(host, device, dc.GetMountPoint(),
+				dc.GetContainerImage(), dc.GetFormatPercent()); err != nil {
+				return err
+			}
+		}
+	}
+
+	// sync disk records
+	diskRecords := curveadm.DiskRecords()
```

Comment:
I think the below logic is a bit complex, you can provide a function to get the difference bettwen new records and old records, then to handle the difference, like receive the commit or denied. 
Maybe you can refer [checkScaleOutTopology](https://github.com/opencurve/curveadm/blob/develop/cli/command/scale_out.go#L184), and if you refuse the commit, you can provide some more meaningful error codes, like `ERR_DELETE_BINDING_DISK_IS_DENIED`.

Reply:
Done.